warp10/src/main/java/io/warp10/script/functions/HHCODETO.java

```diff
@@ -52,17 +53,21 @@ public Object apply(WarpScriptStack stack) throws WarpScriptException {
         hhcode = new StringBuilder(hhstr).append("0000000000000000");
         ((StringBuilder) hhcode).setLength(16);
       }
-      long hh = new BigInteger(hhcode.toString(),16).longValue();
-      latlon = GeoXPLib.fromGeoXPPoint(hh);
+      hh = Long.parseUnsignedLong(hhcode.toString(), 16);
     } else if (hhcode instanceof byte[]) {
-      long hh = Longs.fromByteArray((byte[]) hhcode); 
-      latlon = GeoXPLib.fromGeoXPPoint(hh);
+      hh = Longs.fromByteArray((byte[]) hhcode);
     } else {
       throw new WarpScriptException(getName() + " expects a long, a string or a byte array.");
     }
-    
-    stack.push(latlon[0]);
-    stack.push(latlon[1]);
+
+    if (GeoTimeSerie.NO_LOCATION == hh) {
```

Comment:
This changes the behavior, HHCodes can be used in other contexts than GTS, so the special value NO_LOCATION should not lead to NaN NaN when using HHCODETO. Maybe create a variation of this function which produces NaN NaN for NO_LOCATION, but is called differently.